src/full_client.toit

```diff
@@ -325,11 +325,48 @@ interface PersistenceStore:
   */
   do [block] -> none
 
+class PersistentPacket_:
+  topic /string
+  payload /ByteArray
+  retain /bool
+
+  constructor .topic .payload --.retain:
+
 /**
 A persistence store that stores the packets in memory.
 */
 class MemoryPersistenceStore implements PersistenceStore:
-  UNIMPLEMENTED ::= "UNIMPLEMENTED"
+  storage_ /Map := {:}
+  id_ /int := 0
+
+  store topic/string payload/ByteArray --retain/bool -> int:
+    id := id_++
+    storage_[id] = PersistentPacket_ topic payload --retain=retain
+    return id
+
+  get persistent_id/int [block] [--if_absent] -> none:
+    stored := storage_.get persistent_id
+    if stored:
+      block.call stored.topic stored.payload stored.retain
+    else:
+      if_absent.call persistent_id
+
+  remove persistent_id/int -> none:
+    storage_.remove persistent_id
+
+  /**
+  Calls the given block for each stored packet.
```

Comment:
I assume it is important that these are called in insertion order. That should probably be mentioned.

Reply:
Added a comment to the PersistenceStore interface.